geoproxy/main.go

```diff
@@ -95,42 +113,117 @@ func parseRequestBody(request *http.Request) requestPayloadStruct {
 	return requestPayload
 }
 
-// Log the typeform payload and redirect url
-func logRequestPayload(requestionPayload requestPayloadStruct, proxyUrl string) {
-	log.Printf("lat: %s, lng: %s, proxy_url: %s\n", requestionPayload.Lat, requestionPayload.Lng, proxyUrl)
+// Parse the submit requests body
+func parseSubmitRequestBody(request *http.Request) submitRequestPayloadStruct {
+	decoder := requestBodyDecoder(request)
+
+	var requestPayload submitRequestPayloadStruct
+	err := decoder.Decode(&requestPayload)
+
+	if err != nil {
+		panic(err)
+	}
+
+	return requestPayload
 }
 
-// Get the url for a given lat and long pair
-func getProxyUrl(latRaw string, lngRaw string) string {
-	lat1, err1 := strconv.ParseFloat(latRaw, 32)
-	lng1, err2 := strconv.ParseFloat(lngRaw, 32)
-	lat := float32(lat1)
-	lng := float32(lng1)
+// Log the view typeform payload and redirect url
+func logViewRequestPayload(requestionPayload viewRequestPayloadStruct, proxyUrl string) {
+	log.Printf("latlng1: %s, latlng2: %s, proxy_url: %s\n", requestionPayload.LatLng1, requestionPayload.LatLng2, proxyUrl)
+}
 
-	default_condtion_url := os.Getenv("DEFAULT_CONDITION_URL")
+// Log the submit typeform payload and redirect url
+func logSubmitRequestPayload(requestionPayload submitRequestPayloadStruct, proxyUrl string) {
+	log.Printf("coordinate: %s, proxy_url: %s\n", requestionPayload.LatLng, proxyUrl)
+}
 
-	if err1 != nil || err2 != nil {
-		log.Printf("Error: Invalid lat-long pair.")
-		return default_condtion_url
+// Given string in form [0.0, 0.0], create and return coord struct
+func parseCoord(coord string) Coord {
+	splitCoord := strings.Split(coord, ",")
+	// We should have someething like {"[0.0", "0.0]"}
+	if len(splitCoord) != 2 {
+		// Returning error coord of 360, 360 for now
+		log.Printf("Error: Coordinate passed into json request should be of form [0.0, 0.0].")
+		return Coord{360, 360}
+	} else {
```

Comment:
Maybe just add a comment here. "We get a valid string, but now have to parse out the numbers and make sure that they're valid"